### PR TITLE
 API Rename TravisCheck to CIPassingCheck, implement CircleCI fallback for CI checks

### LINKED ISSUE #1: Abstract HTTP requests out so the the API based checks can be mocked
Ideally we can mock HTTP requests to APIs for the Travis, Scrutinizer and Codecov.io based checks.

This would allow us to properly unit test the checks.

### LINKED ISSUE #4: Add CircleCI as a secondary Travis provider if the Travis check returns 404
CircleCI is probably the next most popular CI provider for running tests on SilverStripe modules.

We could rename the TravisCheck to CiBuildsCheck and have it check Travis first then Circle afterwards if Travis returns a 404

### LINKED ISSUE #6: Add Scrutinizer CI as a fallback for code coverage checks if Codecov.io is not configured
Similar to #4 we should check for Scrutinizer CI code coverage reports if Codecov.io is a 404, since that's probably more common in SilverStripe 3 modules than Codecov which we're suggesting from SS4 onwards.

*Clarification: we'd still recommend using codecov.io on SS3 modules too*

### PR DESCRIPTION
**Pull request by robbieaverill**
Resolves #4 by adding fallback support for CircleCI build checks, resolves #1 by using Guzzle and resolves #6 by falling back to Scrutinizer when Codecov doesn't have a code coverage report.

Also requires https://github.com/silverstripe/moduleratings-plugin/pull/1

### CODE REVIEW

**Comment by Firesphere** [comment]
Looks pretty good to me. Had a look at the Scrutinizer "issues", seems there could be some improvement around inline docks being marked maybe, but that's really minor.

**Comment by Firesphere** [comment]
`->run` gives me a server error, did I miss a configuration setting?

**Comment by robbieaverill** [comment]
Not off the top of my head. How are you running it? Via the composer plugin?

**Comment by Firesphere** [comment]
Composer plugin doesn't work with your fork out of the box, haven't tried that yet. It seems something with getRepositorySlug somewhere down the line breaks for unknown reasons.

File: src/Check/CIPassingCheck.php [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+
+    /**
+     * Uses the Travis API to check whether the latest CI build passed
+     *
+     * @param string $slug
+     * @return bool
+     */
+    protected function checkTravisBuild($slug)
+    {
+        $result = $this->getRequestClient()
```
Comment (Firesphere) [inline src/Check/CIPassingCheck.php]: This request errors when there is no TravisCI build. A check for a travis.yml would fix it. the CircleCI build will probably do the same.

**Comment by Firesphere** [comment]
Hold on, this may be something with my Vagrant box being unable to connect to the API's. I'll have to investigate that tomorrow. Other than that, looks good to me!

**Comment by Firesphere** [comment]
Problem found. It's a confusing one!

Might want to add an update to the readme to state the slug is _case sensitive_, which caused my error

File: src/Check/CIPassingCheck.php (hunk 1 above)
Comment (Firesphere) [reply src/Check/CIPassingCheck.php]: Despite finding what went wrong on my side, I'd still suggest wrapping it in a try/catch block and skip the test if Guzzle throws an error

**Comment by Firesphere** [comment]
Other thing I noticed:
```
      'description' => string 'Has source code in either a "code" or a "src" folder' (length=52)
      'points' => int 0
      'description' => string 'Has a readme file' (length=17)
      'points' => int 0
```

My repo does have a readme file, as well as the code being in `src`, so something might go wrong (unrelated) here?

File: src/Check/CIPassingCheck.php (hunk 1 above)
Comment (robbieaverill) [reply src/Check/CIPassingCheck.php]: Yep, catching errors would make sense. Will update.

**Comment by robbieaverill** [comment]
Re: https://github.com/silverstripe/moduleratings/pull/9#issuecomment-381529749

How are you running this check?

**Comment by Firesphere** [comment]
A quick dump in `PageController::init()`

As per the readme, I have a `Readme` file, which is not one of the options in your test array, probably better to have as a separate PR.

**Comment by robbieaverill** [comment]
Oh right, like this one 😆 https://github.com/silverstripe/moduleratings/pull/10

**Comment by Firesphere** [comment]
👍

**Pull request merged**